polkadot/runtime/parachains/src/paras_inherent/tests.rs

```diff
@@ -2024,7 +2022,21 @@ mod enter {
 			// * 5 v2 candidate descriptors.
 			assert_eq!(inherent_data.backed_candidates.len(), 5);
 
-			Pallet::<Test>::enter(frame_system::RawOrigin::None.into(), inherent_data).unwrap();
+			let mut expected_inherent_data = inherent_data.clone();
+			expected_inherent_data.backed_candidates.truncate(3);
+
+			let mut create_inherent_data = InherentData::new();
+			create_inherent_data
+				.put_data(PARACHAINS_INHERENT_IDENTIFIER, &inherent_data)
+				.unwrap();
+
+			assert_eq!(
```

Comment:
What are we testing here?

Reply:
that indeed the 2 candidates are dropped (for paraid 3 and 4).
We should check the whole inherent data though, I'll do that. I modified this while debugging